### PR TITLE
[scripts] Use build_cxx_project.sh for build_sc_component_manager.sh

### PR DESCRIPTION
**Pull request by NikitaZotov**
* [ ] Update changelog (no needed)
* [ ] Update SCn documentation if needed

Dependencies:
- https://github.com/ostis-ai/sc-machine/pull/372

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,64 +5,53 @@ set(CMAKE_CXX_STANDARD 17)
...
-    set(CMAKE_LIBRARY_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${SC_BIN_PATH})
-endforeach(OUTPUT_CONFIG)
+set(SC_MEMORY_SRC "${SC_MACHINE_PATH}/sc-memory/")
+set(SC_KPM_SRC "${SC_MACHINE_PATH}/sc-kpm/")
+set(SC_TOOLS_SRC "${SC_MACHINE_PATH}/sc-tools/")
+
+include_directories(${SC_MEMORY_SRC} ${SC_KPM_SRC} ${SC_TOOLS_SRC})
+
+if(${WIN32})
+    message(SEND_ERROR "SC-component-manager isn't supported on Windows OS.")
```
Comment (MksmOrlov) [inline CMakeLists.txt:36]:
```suggestion
    message(SEND_ERROR "sc-component-manager isn't supported on Windows OS.")
```

File: scripts/build_sc_component_manager.sh [hunk 2]
```diff
@@ -4,70 +4,16 @@ set -eo pipefail
...
 
-cd "${SC_COMPONENT_MANAGER_CMAKE_PATH}"
-cmake -B "${SC_COMPONENT_MANAGER_BUILD_PATH}" "${SC_COMPONENT_MANAGER_CMAKE_PATH}" ${build_release:+${release_mode}} "${outer_cmake_args[@]}"
-cmake --build "${SC_COMPONENT_MANAGER_BUILD_PATH}" -j"$(nproc)"
+BUILD_SCRIPTS="${SC_COMPONENT_MANAGER_CMAKE_PATH}/../sc-machine/scripts/build-scripts"
+"${BUILD_SCRIPTS}/build_cxx_project.sh" -p "${SC_COMPONENT_MANAGER_CMAKE_PATH}" -s "${SC_COMPONENT_MANAGER_CXX_SOURCES_PATH}" \
+  -b "${SC_COMPONENT_MANAGER_BUILD_PATH}" --cmake-arg "-DSC_BIN_PATH=${BINARY_PATH}" "$@"
 
-stage "Sc-component-manager built successfully"
+stage "SC-component-manager built successfully"
```
Comment (MksmOrlov) [inline scripts/build_sc_component_manager.sh:19]:
```suggestion
stage "sc-component-manager built successfully"
```

**Review by MksmOrlov — CHANGES_REQUESTED** [review]

File: scripts/build_sc_component_manager.sh (hunk 2 above)
Comment (MakarenkoAI) [reply scripts/build_sc_component_manager.sh:19]: _is_ built, and why "sc"?

**Review by MakarenkoAI — CHANGES_REQUESTED** [review]
image
sc-machine and manager : feat/improve_build
reload cmake, build
then reload cmake in manager

File: CMakeLists.txt [hunk 3]
```diff
@@ -5,64 +5,53 @@ set(CMAKE_CXX_STANDARD 17)
...
+link_directories(${SC_BIN_PATH} ${SC_EXTENSIONS_DIRECTORY})
 
-include("${SC_MACHINE_ROOT}/cmake/codegen.cmake")
-include("${SC_MACHINE_ROOT}/cmake/tests.cmake")
+set(SC_MACHINE_PATH "${PARENT_DIR}/sc-machine/")
+include("${SC_MACHINE_PATH}/cmake/codegen.cmake")
 set(SC_CODEGEN_TOOL "${SC_BIN_PATH}/sc-code-generator")
 
-option(SC_CLANG_FORMAT_CODE "Flag to add clangformat and clangformat_check targets" OFF)
+if(${SC_BUILD_TESTS})
```
Comment (MakarenkoAI) [inline CMakeLists.txt:21]: why test.cmake included twice, here and on the 56 line

Comment (MakarenkoAI) [inline CMakeLists.txt:21]: why test.cmake included twice, here and on the 56 line

**Review by MakarenkoAI — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 3 above)
Comment (NikitaZotov) [reply CMakeLists.txt:21]: Firstly, we include cmake for configure any tests. Secondly, we include cmake for some tests. There are different cmakes

File: README.md [hunk 4]
```diff
@@ -1,15 +1,15 @@
-# SC-component-manager
+# sc-component-manager
 
-**SC-component-manager** is a part of [**OSTIS Platform**](https://github.com/ostis-ai/ostis-web-platform).
+This repository is a storage of the Component manager of the [OSTIS Project](https://github.com/ostis-ai/ostis-project).
```
Comment (MksmOrlov) [inline README.md:3]: image

File: README.md [hunk 5]
```diff
@@ -62,7 +62,7 @@ Possible run flags:
 
 File specification.scs contains description of two sections: **components** and **repositories**.
 
-**Repositories** have links for source (GitHub, google drive etc.) with specification file, **components** have links to source with specification of component.
+**Repositories** have links for source (GitHub, Google Drive etc.) with specification file, **components** have links to source with specification of component.
```
Comment (MksmOrlov) [inline README.md:65]:
```suggestion
**Repositories** have links for source on GitHub with specification file, **components** have links to source with specification of component.
```

**Review by MksmOrlov — CHANGES_REQUESTED** [review]

**Review by MksmOrlov — CHANGES_REQUESTED** [review]
Update codegen path in manager cmake to find only in keynodes directory

**Review by MakarenkoAI — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1,15 +1,15 @@
-# SC-component-manager
+# sc-component-manager
 
-**SC-component-manager** is a part of [**OSTIS Platform**](https://github.com/ostis-ai/ostis-web-platform).
+This repository is a storage of the Component manager of the [OSTIS Project](https://github.com/ostis-ai).
```
Comment (MksmOrlov) [inline README.md:3]: I think, repository is a storage itself and there is no need to duplicate it info.

**Review by MksmOrlov — CHANGES_REQUESTED** [review]

**Review by MksmOrlov — APPROVED** [review]

**Pull request merged**